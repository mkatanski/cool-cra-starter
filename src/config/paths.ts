export class Paths {
  // public
  readonly login = "/login";
  readonly empty = "/empty";

  // private
  readonly dashboard = "/dashboard";
}

export const paths = new Paths();
