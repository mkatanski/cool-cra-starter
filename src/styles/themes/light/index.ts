export * from "./main";
export * from "./topBar";
