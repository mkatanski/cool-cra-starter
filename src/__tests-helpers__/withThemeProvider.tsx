import React from "react";
import { ThemeContext } from "components/theme/ThemeProvider";
import { StyledTheme } from "styles/_types";
import { ThemeName } from "config/themes";
import { ThemeProvider } from "styled-components";
import { palette } from "styles/palette";

const TestTheme: StyledTheme = {
  container: {
    background: "container.background",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  button: {
    text: {
      default: "palette.holly",
      hover: "palette.outerSpace",
    },
    background: {
      default: "palette.azalea",
      hover: "palette.charm",
    },
  },

  input: {
    background: {
      default: "palette.azalea",
      hover: "palette.azalea",
    },
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  table: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableHeader: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowEven: {
    background: "palette.wewak",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowOdd: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },
};

export const withThemeProvider = (Component: React.ReactNode) => (
  <ThemeContext.Provider
    value={{
      themeMode: ThemeName.Light,
      changeTheme: jest.fn(),
    }}
  >
    <ThemeProvider
      theme={{
        elements: TestTheme,
        palette: palette,
      }}
    >
      {Component}
    </ThemeProvider>
  </ThemeContext.Provider>
);
