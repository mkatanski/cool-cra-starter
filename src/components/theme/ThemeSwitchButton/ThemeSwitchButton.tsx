import React, { useContext } from "react";
import { ThemeContext } from "../ThemeProvider";
import { ThemeName } from "config/themes";
import { LinkButton } from "components/LinkButton";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export type ThemeSwitchButtonProps = {};

export const ThemeSwitchButton: React.FC<ThemeSwitchButtonProps> = () => {
  const { themeMode, changeTheme } = useContext(ThemeContext);

  const handleClick = () => {
    if (themeMode === ThemeName.Light) {
      changeTheme(ThemeName.Dark);
    } else {
      changeTheme(ThemeName.Light);
    }
  };

  return (
    <LinkButton data-testid="ThemeToggleButton" onClick={handleClick}>
      <FontAwesomeIcon icon="adjust" />
    </LinkButton>
  );
};
