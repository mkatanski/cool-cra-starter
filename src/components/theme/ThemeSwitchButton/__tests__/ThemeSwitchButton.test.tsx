import React, { useContext } from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import { ThemeSwitchButton } from "../ThemeSwitchButton";
import { withThemeProvider } from "__tests-helpers__/withThemeProvider";
import { ThemeName } from "config/themes";

import { render, fireEvent, getByTestId } from "@testing-library/react";

jest.mock("react", () => {
  const ActualReact = require.requireActual("react");
  return {
    ...ActualReact,
    useContext: jest.fn(),
  };
});

const useContextMock = useContext as jest.Mock;

describe("components", () => {
  describe("ThemeSwitchButton", () => {
    it("should render ThemeSwitchButton component", () => {
      const handleToogleClick = jest.fn();
      useContextMock.mockReturnValue({
        themeMode: ThemeName.Light,
        changeTheme: handleToogleClick,
      });

      const tree = TestRenderer.create(
        withThemeProvider(<ThemeSwitchButton />)
      );

      expect(tree).toMatchSnapshot();
    });

    it("should call themeToggle method", () => {
      const handleToogleClick = jest.fn();
      useContextMock.mockReturnValue({
        themeMode: ThemeName.Light,
        changeTheme: handleToogleClick,
      });

      const { container } = render(withThemeProvider(<ThemeSwitchButton />));

      fireEvent.click(getByTestId(container, "ThemeToggleButton"));
      expect(handleToogleClick).toHaveBeenCalledTimes(1);
    });
  });
});
