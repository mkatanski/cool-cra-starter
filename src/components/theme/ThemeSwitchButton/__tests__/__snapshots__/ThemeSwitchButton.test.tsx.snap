// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ThemeSwitchButton should render ThemeSwitchButton component 1`] = `
.c0 {
  background: palette.azalea;
  color: palette.holly;
  cursor: pointer;
  padding: 10px;
  display: inline-block;
  text-align: center;
}

.c0:hover {
  background: palette.charm;
  color: palette.outerSpace;
}

<a
  className="c0"
  data-testid="ThemeToggleButton"
  onClick={[Function]}
>
  <span
    data-props={
      Object {
        "icon": "adjust",
      }
    }
    id="FontAwesomeIcon"
  />
</a>
`;
